exir/backend/utils.py

```diff
@@ -569,3 +572,81 @@ def __call__(self, node: torch.fx.Node, reason: str) -> None:
 
     def __str__(self) -> str:
         return f"WhyNoPartition: Node {self.node} was not partitioned because {self.reason}."
+
+
+class WhyNoPartitionReporter:
+    """
+    Helper class for partitioners to gather why nodes were not lowered in a single report.
+    If a node is reported multiple times, only the first report is included.
+
+    Example usage:
+
+        # In your backend partitioner file(s)
+        reporter = WhyNoPartitionReporter()
+
+        # hypothetical function that checks if a node can be lowered
+        if not can_be_lowered(node):
+            reporter.report_reject(node, "This node was not lowered because ...")
+
+        # Back in partitioner
+        logger.info(reporter.get_table_report())
+    """
+
+    def __init__(self):
+        self._rejected_nodes: OrderedDict[torch.fx.Node, str] = (
+            OrderedDict()
+        )  # {Rejected node: reason}
+
+    def report_reject(self, node: torch.fx.Node, reason: str):
+        if node not in self._rejected_nodes:
+            self._rejected_nodes[node] = reason
+
+    def get_table_report(self) -> str:
+        reject_report = []
+        for node in self._rejected_nodes:
+            if node.op == "placeholder" or node.op == "output":
+                continue
+            if not (target := getattr(node.target, "_op", None)):
+                target = node.target
+            torch_fn = node.meta.get("torch_fn", "-")
+            reject_report.append(
+                [node.name, target, torch_fn, self._rejected_nodes[node]]
+            )
+        if len(reject_report) > 0:
+            return (
+                tabulate(
+                    reject_report,
+                    ["Name", "Target", "Torch func", "Reason"],
+                    tablefmt="fancy_grid",
+                    maxcolwidths=35,
+                )
+                + "\n(Placeholders and outputs are not included in this list)"
+            )
+        else:
+            return "Partitioned all operators."
```

Comment:
Perhaps better to say "No Operators Rejected" Since partitioner doesn't necessarily partition all operators in this case.